cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,36 @@
 /// <reference types='cypress' />
-
-describe('Student Registration page', () => {
-  before(() => {
-
-  });
-
-  it('', () => {
-
+describe('Automation Practice Form', () => {
+  it('should successfully register a new user with valid data', () => {
+    cy.visit(Cypress.env('form_url'));
+    cy.get('#firstName').type('John');
+    cy.get('#lastName').type('Doe');
+    cy.get('#userEmail').type('john.doe@example.com');
+    cy.get('label[for="gender-radio-1"]').click();
+    cy.get('#userNumber').type('1234567890');
+    cy.get('#dateOfBirthInput').click();
+    cy.get('.react-datepicker__month-select').select('April');
+    cy.get('.react-datepicker__year-select').select('1990');
+    cy.get('.react-datepicker__day--015').click();
+    cy.get('#subjectsInput').type('Math{enter}');
```

Comment:
It's a good practice to clear an input field before typing into it. This ensures that the test doesn't fail due to some pre-filled value. You can use `.clear()` before `.type()`.